mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala

```diff
@@ -197,14 +240,14 @@ private[ml] object SummaryBuilderImpl extends Logging {
    * metrics that need to de computed internally to get the final result.
    */
   private val allMetrics: Seq[(String, Metric, DataType, Seq[ComputeMetric])] = Seq(
-    ("mean", Mean, arrayDType, Seq(ComputeMean, ComputeWeightSum)),
-    ("variance", Variance, arrayDType, Seq(ComputeWeightSum, ComputeMean, ComputeM2n)),
+    ("mean", Mean, vectorUDT, Seq(ComputeMean, ComputeWeightSum)),
+    ("variance", Variance, vectorUDT, Seq(ComputeWeightSum, ComputeMean, ComputeM2n)),
     ("count", Count, LongType, Seq()),
-    ("numNonZeros", NumNonZeros, arrayLType, Seq(ComputeNNZ)),
-    ("max", Max, arrayDType, Seq(ComputeMax, ComputeNNZ)),
-    ("min", Min, arrayDType, Seq(ComputeMin, ComputeNNZ)),
-    ("normL2", NormL2, arrayDType, Seq(ComputeM2)),
-    ("normL1", NormL1, arrayDType, Seq(ComputeL1))
+    ("numNonZeros", NumNonZeros, vectorUDT, Seq(ComputeNNZ)),
```

Comment:
Could you let me know why did you make this change? I think we should use long array rather than double array to store ```numNonZeros```.

Reply:
`org.apache.spark.mllib.stat.MultivariateOnlineSummarizer` also return `Vector` for `numNonZeros`. So I prefer keep consistent with it.

Reply:
In the old ```mllib.stat.MultivariateOnlineSummarizer```, the internal variable is type of ```Array[Long]```, but the return type is ```Vector```. Do you know the impact of using ```Vector``` internal? Thanks.

Reply:
Internally still use `Array[Long]` to do the computation. Only when returning result, convert it to vector.